jax/_src/deprecations.py

```diff
@@ -57,11 +57,18 @@ def getattr(name):
 
 
 def accelerate_module_deprecation(module: ModuleType, name: str) -> None:
-  """Accelerate the deprecation of a module-level attribute"""
+  """Accelerate the deprecation of a module-level attribute.
+
+  Raises an AttributeError instead of a DeprecationWarning upon attribute access.
+  Used in Google-internal code to implement faster deprecation.
+  """
   message, _ = module._deprecations[name]
   module._deprecations[name] = (message, None)
 
 
+# Maps (module, attr) to False if the attribute is deprecated and should emit
+# a DeprecationWarning and to True if it is deprecated and should raise an
+# AttributeError.
```

Comment:
I am not convinced if this is correct. Is it?

Reply:
Yes, but a shorter way of saying this is that it maps a pair of strings to a boolean specifying whether the deprecation is accelerated. The intent is that non-accelerated deprecations will warn, and accelerated deprecations will error.

Reply:
I found the API to be confusing, or perhaps I do not understand how to use it. Most modules use `__getattr__ = _deprecation_getattr(__name__, _deprecations)` as documented at the start of `deprecations.py`. It seems to be important to use the special name `_deprecations`, if one wants to call `accelerate_deprecation`.

However, there is a whole other set of APIs to register and accelerate a deprecation, that seems to have nothing to do with the `_deprecation_getattr` API.